pkg/build/controller/common/util.go

```diff
@@ -134,3 +140,61 @@ func HasBuildPodNameAnnotation(build *buildapi.Build) bool {
 	_, hasAnnotation := build.Annotations[buildapi.BuildPodNameAnnotation]
 	return hasAnnotation
 }
+
+// ProcessValueFrom supports the k8s pod syntax of allowing "valueFrom" in addition
+// to "value", where "valueFrom" uses the downward api to reference existing env
+// variables, jsonpath references to the build object, secrets, and configmaps.
+func ProcessValueFrom(build *buildapi.Build, client kclientset.Interface) (*[]kapi.EnvVar, error) {
```

Comment:
given that this is a return value and not a parameter, it should be entirely possible to just return []Foo and let the caller look up the pointer if it wants to use it.